modern/mdevent/dispatcher.go

```diff
@@ -0,0 +1,75 @@
+package mdevent
+
+import (
+	"github.com/short-d/app/fw"
+	"sync"
+
+	"github.com/asaskevich/EventBus"
+)
+
+var _ fw.Dispatcher = (*EventDispatcher)(nil)
+
+// EventDispatcher publishes an event to all its subscribers
+type EventDispatcher struct {
+	eventBus EventBus.Bus
```

Comment:
Let's add `EventBus` interface so that we have opportunity to swap it out for testing or replace it with a different library.

Reply:
`EventBus.Bus` is already an interface. I really don't see here any point to add our own `EventBus` interface. There are some of my thoughts:

* If we had to replace one library with another, we would create an adapter for our `EventBus` interface. It's redundant stuff in my opinion.
* `Dispatcher` declared as a composition of interfaces, it means in the majority cases, there will be the only dependency on `Emitter` interface (which consist of the single method). We can easily replace `Emitter` for tests.

Please correct me, if I am wrong.

Reply:
[This](https://github.com/short-d/app/blob/620b15942501a3b06661f98171c5cb8163cd816d/tool/terminal/terminal.go#L79)  also uses eventbus. They should be unified in the future.